File: backend/src/main/java/com/cmg/comtogether/product/entity/Product.java
package com.cmg.comtogether.product.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {

    @Id
    private String productId;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String link;

    private String image;

    private Integer lprice;
    private Integer hprice;

    @Column(nullable = false)
    private String mallName;

    @Column(nullable = false)
    private String productType;

    private String maker;
    private String brand;

    private String category1;
    private String category2;
    private String category3;
    private String category4;
}

File: frontend/src/types/product.ts
// 제품 관련 타입 정의

export interface Product {
  product_id: string;
  title: string;
  lprice: string;
  hprice: string;
  image: string;
  link: string;
  mall_name: string;
  product_type: string;
  maker: string;
  brand: string;
  category1: string;
  category2: string;
  category3: string;
  category4: string;
}

export interface ProductListResponse {
  total: number;
  start: number;
  display: number;
  items: Product[];
}

export interface GetProductsParams {
  category: string;
  query?: string;
  display?: number;
  start?: number;
  sort?: 'sim' | 'date' | 'asc' | 'dsc';
  exclude?: string;
  [key: string]: unknown;
}

export interface GetRecommendedProductsParams {
  category: string;
  query?: string;
  display?: number;
  start?: number;
  sort?: 'sim' | 'date' | 'asc' | 'dsc';
  exclude?: string;
  [key: string]: unknown;
}
